### ISSUE TITLE
Lifetime error when using nested views

### ISSUE THREAD

**Issue by jafioti** [sycamore-rs/sycamore#387]
When using nested views and an action callback, a lifetime error is thrown.

Open any sycamore project and paste:
```rust
#[component]
pub fn TempComponent<G: Html>(ctx: Scope) -> View<G> {
    let action = |_| ctx.spawn_local(async move {});
    view!{ctx, 
        ({
            view!{ctx, 
                input (on:click=action)
            }
        })
    }
}
```

Expected that no lifetime error is thrown since this is valid code.

- Sycamore: master

**Comment by lukechu10** [comment]
This regression was introduced by https://github.com/sycamore-rs/sycamore/commit/9fdff59b9c0c399cd8793ae73fe7fd1e7d7ec3a3 because it adds an addition `create_effect_scoped` call. 

Unfortunately, this change is necessary to prevent memory leaks and it seems like it interferes with Rust's borrow checker. I suspect it has something to do with the invariance of the lifetime parameter on `Scope` because accessing a signal or other from inside the closure works.

In other words, the following works:
```rust
#[component]
pub fn TempComponent<G: Html>(ctx: Scope) -> View<G> {
    let action = |_| {
        ctx.create_signal(0);
    };
    view!{ ctx, 
        ({
            view!{ ctx, 
                input (on:click=action)
            }
        })
    }
}
```
I'm guessing because `&'a Signal<i32>` is covariant over `'a`.

If somebody can figure this out, that would be greatly appreciated!

**Comment by lukechu10** [comment]
But for your specific use case, since the interpolated value is not actually reactive, you can just create the inner view outside of the outer view like so:
```rust
#[component]
pub fn TempComponent<G: Html>(ctx: Scope) -> View<G> {
    let action = |_| ctx.spawn_local(async move {});
    let inner = view!{ ctx, 
        input (on:click=action)
    };
    view!{ ctx, 
        (inner)
    }
}
```

**Comment by lukechu10** [comment]
Eureka! This is because of rustc non-lexical-lifetimes "migration mode". If you add `#![feature(nll)]` with nightly toolchain, the issue is fixed!

**Comment by lukechu10** [comment]
Rust tracking issue for turning off nll migration mode: https://github.com/rust-lang/rust/issues/58781
